test/test_optim.py

```diff
@@ -567,6 +536,7 @@ def test_old_pattern_warning_resuming(self):
 
         with warnings.catch_warnings(record=True) as ws:
             warnings.simplefilter("always")  # allow any warning to be raised
+            warnings.simplefilter("ignore", category=DeprecationWarning)
```

Comment:
I don't see why you need this ignore here. We're not stepping, right?

Reply:
(I won't comment on the other occurrences of this.)

Reply:
In the tests referred, __init__ is executing step with a nonzero initial step which triggers the warning.